nexa/gguf/streamlit/streamlit_image_chat.py

```diff
@@ -37,93 +105,236 @@ def generate_images(nexa_model: NexaImageInference, prompt: str, negative_prompt
+            \n2. Find a vision model using the task filters
+            \n3. Select your desired model and copy either:
+            \n   - The full nexa run command, or (e.g., nexa run stable-diffusion-v1-4:q4_0)
+            \n   - Simply the model name (e.g., stable-diffusion-v1-4:q4_0)
+            \n4. Paste it into the field on the sidebar and press enter
+            """)
+            st.stop()
+
+        # process the input after checking it's not empty:
+        if hub_model_name.startswith("nexa run"):
+            hub_model_name = hub_model_name.split("nexa run")[-1].strip()
+        else:
+            hub_model_name = hub_model_name.strip()
+
+        # handle hub model name changes:
+        if 'nexa_model' not in st.session_state or st.session_state.current_hub_model != hub_model_name:
+            with st.spinner("Loading model from hub..."):
+                st.session_state.nexa_model = load_model(hub_model_name)
+                if st.session_state.nexa_model:  # only update if load was successful
+                    st.session_state.current_hub_model = hub_model_name
 
-if st.button("Generate Image"):
-    if not prompt:
-        st.warning("Please enter a prompt to proceed.")
     else:
-        with st.spinner("Generating images..."):
-            images = generate_images(
-                st.session_state.nexa_model, prompt, negative_prompt
-            )
-            st.success("Images generated successfully!")
-            for i, image in enumerate(images):
-                st.image(image, caption=f"Generated Image", use_column_width=True)
-                
-                img_byte_arr = io.BytesIO()
-                image.save(img_byte_arr, format='PNG')
-                img_byte_arr = img_byte_arr.getvalue()
-
-                st.download_button(
-                    label=f"Download Image",
-                    data=img_byte_arr,
-                    file_name=f"generated_image.png",
-                    mime="image/png"
+        # load selected model if it's not already loaded:
+        if ('nexa_model' not in st.session_state or getattr(st.session_state, 'current_model_path', None) != model_path):
+            with st.spinner(f"Loading model: {model_path}"):
+                st.session_state.nexa_model = load_model(model_path)
+                if st.session_state.nexa_model:  # only update if load was successful
+                    st.session_state.current_model_path = model_path
+
+    # generation params:
+    if 'nexa_model' in st.session_state and st.session_state.nexa_model:
+        st.sidebar.header("Generation Parameters")
+
+        model_to_check = (st.session_state.current_hub_model if st.session_state.current_hub_model else st.session_state.current_local_path if st.session_state.current_local_path else st.session_state.current_model_path)
+
+        # get model specific defaults:
+        default_params = get_default_params(model_to_check)
+
+        # adjust step range based on model type:
```

Comment:
You should ready in those parameters from
https://github.com/NexaAI/nexa-sdk/blob/main/nexa/constants.py#L311-L353

use
`DEFAULT_IMG_GEN_PARAMS_LCM.num_inference_steps`

Reply:
@zhiyuan8 Let me clarify the `get_default_params()` approach:
Different computer vision models require their own default parameters. More specifically, 
- LCM models → DEFAULT_IMG_GEN_PARAMS_LCM (4 steps, 1.0 guidance)
- SDXL-turbo → DEFAULT_IMG_GEN_PARAMS_TURBO (5 steps, 5.0 guidance)
- Standard SD models → DEFAULT_IMG_GEN_PARAMS (20 steps, 7.5 guidance)
The `get_default_params` function at https://github.com/NexaAI/nexa-sdk/blob/main/nexa/constants.py#L23-L30 picks the right parameter set based on model name(type). Happy to add clearer documentation, or let me know if I didn't get your point.